### PR TITLE
Inm/#2067 adding param to adjust in demand following repositioning manager how many closest clusters to consider

### PR DESCRIPTION
**Pull request by irishwithaxe**
#2067

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2068)

### CODE REVIEW

**Comment by irishwithaxe** [comment]
Test!

File: src/main/resources/beam-template.conf [hunk 1]
```diff
@@ -134,6 +134,8 @@ beam.agentsim.agents.rideHail.repositioningManager.timeout = "int | 0"
 # Larger value increase probability of the ride-hail vehicle to reposition
 beam.agentsim.agents.rideHail.repositioningManager.demandFollowingRepositioningManager.sensitivityOfRepositioningToDemand = "double | 1"
 beam.agentsim.agents.rideHail.repositioningManager.demandFollowingRepositioningManager.numberOfClustersForDemand = "int | 30"
+beam.agentsim.agents.rideHail.repositioningManager.demandFollowingRepositioningManager.PercentageOfClosestClustersToConsider = "double | 0.2"
```
Comment (colinsheppard) [inline src/main/resources/beam-template.conf]: Please follow convention for naming. Should be camel case with **lower case** for the first letter.

Also, this value is not a percentage, it's a fraction. Should be named "fractionOfClostesClustersToConsider"

File: src/main/scala/beam/agentsim/agents/ridehail/repositioningmanager/DemandFollowingRepositioningManager.scala [hunk 2]
```diff
@@ -130,17 +130,23 @@ class DemandFollowingRepositioningManager(val beamServices: BeamServices, val ri
   private def findWhereToReposition(tick: Int, vehicleLocation: Coord, vehicleId: Id[Vehicle]): Option[Coord] = {
     val currentHour = tick / 3600
     val nextHour = currentHour + 1
+    val percentageOfClosestClusters =
+      beamServices.beamConfig.beam.agentsim.agents.rideHail.repositioningManager.demandFollowingRepositioningManager.PercentageOfClosestClustersToConsider
+
     hourToClusters.lift(nextHour).map { clusters =>
-      // We get top 5 closest clusters and randomly pick one of them.
+      val N = (clusters.length * percentageOfClosestClusters).toInt
```
Comment (colinsheppard) [inline src/main/scala/beam/agentsim/agents/ridehail/repositioningmanager/DemandFollowingRepositioningManager.scala]: Please round to nearest whole number, "toInt" is a "floor" operation and could confuse users

File: src/main/scala/beam/agentsim/agents/ridehail/repositioningmanager/DemandFollowingRepositioningManager.scala [hunk 3]
```diff
@@ -130,17 +130,23 @@ class DemandFollowingRepositioningManager(val beamServices: BeamServices, val ri
...
+      val topNClosest = clusters.sortBy(x => beamServices.geo.distUTMInMeters(x.coord, vehicleLocation)).take(N)
+      val pmf = topNClosest.map { x =>
         new CPair[ClusterInfo, java.lang.Double](x, x.size.toDouble)
       }.toList
+
       val distr = new EnumeratedDistribution[ClusterInfo](rng, pmf.asJava)
       val sampled = distr.sample()
       logger.debug(
-        s"tick $tick, currentHour: $currentHour, nextHour: $nextHour, vehicleId: $vehicleId, vehicleLocation: $vehicleLocation. Top 5 closest: ${top5Closest.toVector}, sampled: $sampled"
+        s"tick $tick, currentHour: $currentHour, nextHour: $nextHour, vehicleId: $vehicleId, vehicleLocation: $vehicleLocation. Top 5 closest: ${topNClosest.toVector}, sampled: $sampled"
```
Comment (colinsheppard) [inline src/main/scala/beam/agentsim/agents/ridehail/repositioningmanager/DemandFollowingRepositioningManager.scala]: Change "Top 5" in this string to the actual number being used, e.g.: ("Top {}",N)

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by colinsheppard** [comment]
test!

**Comment by wrashid** [comment]
Test!

**Pull request merged**